comp/dogstatsd/server/component.go

```diff
@@ -57,3 +56,7 @@ func MockModule() fxutil.Module {
 	return fxutil.Component(
 		fx.Provide(newMock))
 }
+
+func (s *server) Stop() {
```

Comment:
Let's move this function to it's own file [similar to what we do with the logs agent here](https://github.com/DataDog/datadog-agent/blob/23e7aecadca58ae0b6f8264e28c4b0b63aa51771/comp/logs/agent/serverless.go#L37)